packages/api/src/user/services/users.service.ts

```diff
@@ -305,17 +304,13 @@ export const findUsersToEmail = (
           },
         },
       ],
-      ...(partner
-        ? { mail: { $regex: PARTNERS_EMAIL_DOMAIN[partner], $options: "i" } }
-        : {
-            $and: Object.values(PARTNERS_EMAIL_DOMAIN).map((domain: string) => {
-              return {
-                mail: {
-                  $not: { $regex: domain, $options: "i" },
-                },
-              };
-            }),
-          }),
+      $and: Object.values(PARTNERS_EMAIL_DOMAIN).map((domain: string) => {
```

Comment:
On peut pas directement faire un join ou une regex dédiée dans une variable hors de la requete ? 
Le $and je suis pas sur qu'il soit indispensable. Juste la condition "mail: {" est suffisante il me semble